src/main/java/org/example/cursework2/service/impl/JavaQuestionService.java

```diff
@@ -0,0 +1,58 @@
+package org.example.cursework2.service.impl;
+
+import org.example.cursework2.Model.Question;
+import org.example.cursework2.exception.QuestionAlreadyExistsException;
+import org.example.cursework2.exception.QuestionAreEmptyException;
+import org.example.cursework2.exception.QuestionNotFoundException;
+import org.example.cursework2.service.QuestionService;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+@Service
+public class JavaQuestionService implements QuestionService {
+
+    public final Set<Question> questions = new HashSet<>();
+    private final Random random = new Random();
+
+    @Override
+    public Question add(String question, String answer) {
+        return add(new Question(question, answer));
+    }
+
+    @Override
+    public Question add(Question question) {
+        if (questions.add(question)) {
```

Comment:
если успешно добавили, то кидаем исключение? Нет, должно быть наоборот отрицание в условии
```suggestion
        if (!questions.add(question)) {
```